src/Spice86/ViewModels/DisassemblyViewModel.cs

```diff
@@ -0,0 +1,153 @@
+
+    [ObservableProperty]
+    private AvaloniaList<CpuInstructionInfo> _instructions = new();
+    
+    [ObservableProperty]
+    [NotifyCanExecuteChangedFor(nameof(StepInstructionCommand))]
+    [NotifyCanExecuteChangedFor(nameof(ContinueCommand))]
+    private bool _isPaused;
+
+    [ObservableProperty]
+    private bool _isGdbServerAvailable;
+
+    public DisassemblyViewModel() {
+        if (!Design.IsDesignMode) {
+            throw new InvalidOperationException("This constructor is not for runtime usage");
+        }
+    }
+
+    public DisassemblyViewModel(IPauseStatus pauseStatus) {
+        _pauseStatus = pauseStatus;
+        IsPaused = pauseStatus.IsPaused;
+        _pauseStatus.PropertyChanged += OnPauseStatusChanged;
+    }
+
+    private void OnPauseStatusChanged(object? sender, PropertyChangedEventArgs e) {
+        IsPaused = _pauseStatus?.IsPaused is true;
+        if(IsPaused) {
+            _needToUpdateDisassembly = true;
+        }
+    }
+
+    public void Visit<T>(T component) where T : IDebuggableComponent {
+        switch (component) {
+            case IMemory mem:
+                _memory ??= mem;
+                break;
+            case Cpu cpu: {
+                if (_memory is not null && _needToUpdateDisassembly && IsPaused) {
+                    UpdateDisassembly(cpu, _memory);
+                }
+                break;
+            }
+            case IProgramExecutor programExecutor:
+                _programExecutor ??= programExecutor;
+                IsGdbServerAvailable = programExecutor.IsGdbCommandHandlerAvailable;
+                break;
+        }
+    }
+
+    [RelayCommand(CanExecute = nameof(IsPaused))]
+    public void StepInstruction() {
+        _programExecutor?.StepInstruction();
+    }
+
+    [RelayCommand]
+    public void Pause() {
+        if (_programExecutor is null || _pauseStatus is null) {
+            return;
+        }
+        _pauseStatus.IsPaused = _programExecutor.IsPaused = IsPaused = true;
+    }
+
+    [RelayCommand(CanExecute = nameof(IsPaused))]
+    public void Continue() {
+        if (_programExecutor is null || _pauseStatus is null) {
+            return;
+        }
+        _pauseStatus.IsPaused = _programExecutor.IsPaused = IsPaused = false;
+    }
+
+    private void UpdateDisassembly(Cpu cpu, IMemory memory) {
+        _needToUpdateDisassembly = false;
+        uint currentIp = cpu.State.IpPhysicalAddress;
+        CodeReader codeReader = CreateCodeReader(memory, out EmulatedMemoryStream emulatedMemoryStream);
+        Decoder decoder = InitializeDecoder(codeReader, currentIp);
+        try {
+            DecodeFiftyInstructions(cpu, memory, emulatedMemoryStream, currentIp, decoder);
+        } finally {
+            emulatedMemoryStream.Dispose();
+        }
+    }
+
+    // TODO: Infinite scroll of instructions (UI paging)
+    private void DecodeFiftyInstructions(Cpu cpu, IMemory memory, EmulatedMemoryStream emulatedMemoryStream, uint currentIp,
```

Comment:
You probably want to make the amount of decoded instructions configurable

Reply:
Done. The start address is now configurable, too.